File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/daos/ITimerDao.java
package com.dkellycollins.triggerfinger.data.daos;

import com.dkellycollins.triggerfinger.data.entity.ITimer;

/**
 * Provides te ability to manage timer data.
 */
public interface ITimerDao extends IActivityDao {

    Iterable<ITimer> retrieve();
    ITimer retrieve(int id);

    int create(int setTime, int currentTime, boolean isRunning);

    void update(int id, int currentTime, boolean isRunning);

    void delete(int id);
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/daos/ITimerDao.kt
package com.dkellycollins.triggerfinger.data.daos

import com.dkellycollins.triggerfinger.data.entity.ITimer

/**
 * Provides te ability to manage timer data.
 */
interface ITimerDao : IActivityDao {

    fun retrieve(): Iterable<ITimer>
    fun retrieve(id: Int): ITimer

    fun create(setTime: Int, currentTime: Int, isRunning: Boolean): Int

    fun update(id: Int, currentTime: Int, isRunning: Boolean)

    fun delete(id: Int)
}
